Sync and update 'which' across KeyboardEvent, MouseEvent and UIEvent interfaces as per web specification

Ahmad-S792 opened the pull request:
#### df7782c2b48baf7b7f2c5c23912ea4754232fc11
<pre>
Sync and update &apos;which&apos; across KeyboardEvent, MouseEvent and UIEvent interfaces as per web specification

<a href="https://bugs.webkit.org/show_bug.cgi?id=253366">https://bugs.webkit.org/show_bug.cgi?id=253366</a>
<a href="https://rdar.apple.com/problem/106580687">rdar://problem/106580687</a>

Reviewed by Ryosuke Niwa.

This patch aligns WebKit with Gecko / Firefox, Blink / Chromium and Web Specifications [1], [2] and [3].

[1] <a href="https://w3c.github.io/uievents/#idl-uievent">https://w3c.github.io/uievents/#idl-uievent</a>
[2] <a href="https://w3c.github.io/uievents/#idl-keyboardevent">https://w3c.github.io/uievents/#idl-keyboardevent</a>
[3] <a href="https://w3c.github.io/uievents/#idl-mouseevent">https://w3c.github.io/uievents/#idl-mouseevent</a>

This patch modifies &apos;which&apos; to be &apos;unsigned&apos; as per web-specification and also remove it from &apos;KeyboardEvent&apos;.

Merge: <a href="https://chromium.googlesource.com/chromium/src.git/+/cf13270c47f29b21ce7a8c937f598989cd66c840">https://chromium.googlesource.com/chromium/src.git/+/cf13270c47f29b21ce7a8c937f598989cd66c840</a>

* Source/WebCore/dom/KeyboardEvent.cpp:
(KeyboardEvent::which):
* Source/WebCore/dom/KeyboardEvent.h:
* Source/WebCore/dom/KeyboardEvent.idl:
* Source/WebCore/dom/MouseEvent.cpp:
(MouseEvent::which):
* Source/WebCore/dom/MouseEvent.h:
* Source/WebCore/dom/UIEvent.cpp:
(UIEvent::which):
* Source/WebCore/dom/UIEvent.h:
* Source/WebCore/dom/UIEvent.idl: Remove FIXME and align with web-spec

Canonical link: <a href="https://commits.webkit.org/273701@main">https://commits.webkit.org/273701@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/49be593dfa74b6ee0538b97960138e10fd211645

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/36009 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/14954 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/38206 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/38727 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/32390 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/37230 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/17385 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/11984 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/31108 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/36563 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/12650 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/31972 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/11094 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/11103 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/32154 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/39975 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/32716 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/32487 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/37058 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/11277 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/9188 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/35142 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/13019 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/8247 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/11782 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/12115 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

rniwa commented on Source/WebCore/dom/MouseEvent.cpp:
```diff
@@ -194,14 +195,14 @@ MouseButton MouseEvent::button() const
     if (!m_buttonDown)
         return 0;
-    return m_button + 1;
+    return (unsigned)(m_button + 1);
```
Please use `static_cast<unsigned>` instead.

Ahmad-S792 replied:
Done in [49d41ba](https://github.com/WebKit/WebKit/commit/49d41bae5e42c446c7b7eeab027bd5535a50824c).

rniwa commented on Source/WebCore/dom/KeyboardEvent.cpp:
```diff
@@ -232,13 +233,13 @@ bool KeyboardEvent::isKeyboardEvent() const
     if (m_which)
         return m_which.value();
-    return keyCode();
+    return (unsigned)keyCode();
```
Do we really need the cast? If so, please use `static_cast<unsigned>` instead.

Ahmad-S792 replied:
It works without cast but just to be safe and avoid any future compiler or static analysis stuff, I am adding it in [49d41ba](https://github.com/WebKit/WebKit/commit/49d41bae5e42c446c7b7eeab027bd5535a50824c).

Ahmad-S792 committed:
Sync and update 'which' across KeyboardEvent, MouseEvent and UIEvent interfaces as per web specification

https://bugs.webkit.org/show_bug.cgi?id=253366
rdar://problem/106580687

Reviewed by Ryosuke Niwa.

This patch aligns WebKit with Gecko / Firefox, Blink / Chromium and Web Specifications [1], [2] and [3].

[1] https://w3c.github.io/uievents/#idl-uievent
[2] https://w3c.github.io/uievents/#idl-keyboardevent
[3] https://w3c.github.io/uievents/#idl-mouseevent

This patch modifies 'which' to be 'unsigned' as per web-specification and also remove it from 'KeyboardEvent'.

Merge: https://chromium.googlesource.com/chromium/src.git/+/cf13270c47f29b21ce7a8c937f598989cd66c840

* Source/WebCore/dom/KeyboardEvent.cpp:
(KeyboardEvent::which):
* Source/WebCore/dom/KeyboardEvent.h:
* Source/WebCore/dom/KeyboardEvent.idl:
* Source/WebCore/dom/MouseEvent.cpp:
(MouseEvent::which):
* Source/WebCore/dom/MouseEvent.h:
* Source/WebCore/dom/UIEvent.cpp:
(UIEvent::which):
* Source/WebCore/dom/UIEvent.h:
* Source/WebCore/dom/UIEvent.idl: Remove FIXME and align with web-spec

Canonical link: https://commits.webkit.org/273701@main
